projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -555,6 +579,20 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
     return newDate;
   }
 
+  private trySpinValue(datePart: DatePart, targetDatePart: DatePart, spinDeltas: DatePartDeltas, negative = false) {
+    let newValue = null;
+    Object.keys(spinDeltas).forEach(p => {
```

Comment:
Really no need to loop those here. You should be able to extract the delta with just something like `spinDelta[part] || 1`,  right?